src/controllers/usersController.js

```diff
@@ -0,0 +1,52 @@
+const jwt = require('jsonwebtoken');
+const {
+  verifyLoginCredentials,
+  updateUserSession
+} = require('../models/usersModel');
+
+const login = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  await verifyLoginCredentials(email, password)
+    .then((dbResponse) => {
+
+      // Verify email and password
+      if (dbResponse.length !== 1) {
+        // There should only ever be one user with the same email and pw
+        res.status(400).send('Bad Request');
+      }
+      const userID = dbResponse[0].UserID;
+
+      // Create unique session ID (aka jwt)
+      const token = jwt.sign(
+        {
+          data: email
+        },
+        'notASecret',
```

Comment:
Is this a temporary secret? I would suggest putting the secret in the config.js file. Optionally we could also put it in environment variables for better security.

Reply:
So I'm using two different JWT middlewares because jsonwebtoken doesn't work with react. react-jwt doesn't have a way to verify a signature (which is where the secret comes into play), but jsonwebtoken requires a secret. So it's basically just dummy data and doesn't need to be secure. I can still move it to the config file for future proofing, modularity, etc. Good call out!

Reply:
I think the verify() method and the secret can be used in the backend middleware where we want to verify the token when user is trying to access a private endpoint.
And I think you're right for the frontend - we don't need the secret there. From what I know the frontend just needs to store the token, and it'll simply pass the token to the backend in the request header and the backend will verify it.